app/src/main/java/io/spaceapi/community/myhackerspace/Main.java

```diff
@@ -685,22 +707,50 @@ private void populateDataHs() {
             if (data.state != null && data.state.lastchange != null) {
                 final ZonedDateTime lastchange = data.state.lastchange.atZone(ZoneId.systemDefault());
                 final String lastchangeString = lastchange.format(this.dateTimeFormatter);
-                addEntry(iftr, vg, getString(R.string.api_lastchange) + " " + lastchangeString);
+                addEntry(iftr, vg, getString(R.string.api_lastchange, lastchangeString));
             }
 
             // Location
-            addTitle(iftr, vg, R.string.api_location);
-            TextView latLonTv = addEntry(iftr, vg, data.location.lat + ", " + data.location.lon);
-            Linkify.addLinks(
-                latLonTv,
-                Pattern.compile("^.*$", Pattern.DOTALL),
-                String.format(MAP_COORD, data.location.lat, data.location.lon)
-            );
-
-            // Postal address
-            if (data.location.address != null) {
-                addTitle(iftr, vg, R.string.api_postal_addr);
-                addEntry(iftr, vg, data.location.address);
+            if (data.location != null) {
+                addTitle(iftr, vg, R.string.api_location);
+                if (data.location.lat != null && data.location.lon != null) {
+                    TextView latLonTv = addEntry(iftr, vg,
+                        data.location.lat + ", " + data.location.lon);
+                    Linkify.addLinks(
+                        latLonTv,
+                        Pattern.compile("^.*$", Pattern.DOTALL),
+                        String.format(MAP_COORD, data.location.lat, data.location.lon)
+                    );
+                }
+
+                // Postal address
+                if (data.location.address != null) {
+                    addSubtitle(iftr, vg, R.string.api_postal_addr);
+                    addEntry(iftr, vg, data.location.address);
+                }
+                // Country code
+                if (data.location.country_code != null) {
+                    addSubtitle(iftr, vg, R.string.api_location_country_code);
+                    addEntry(iftr, vg, data.location.country_code);
+                }
+                // Hint
+                if (data.location.hint != null) {
+                    addSubtitle(iftr, vg, R.string.api_location_hint);
+                    addEntry(iftr, vg, data.location.hint);
+                }
+                // Areas
+                if (data.location.areas.length > 0) {
```

Comment:
Can't `areas` be `null`?

Reply:
Or are nullable arrays just mapped to empty arrays anyways in case they are `null`?

Reply:
It's an empty array when not present in the API response, and the schema does not declare the key to be nullable. So areas: null means the endpoint is not valid. Should I add a not null check for this key anyway?

Reply:
> It's an empty array when not present in the API response,

Thats what I meant. If it beeing not present in the JSON maps to Java null or empty array.

I guess JSON null would also map to empty array.

Reply:
Ah, true, in the spaceapi-kt library the [areas field is non-nullable but with an empty array as default](https://github.com/spaceapi-community/spaceapi-kt/blob/fc21e4f38a8ca37192d60313a76fb5f89eaac71d/src/main/kotlin/io/spaceapi/types/Location.kt#L36) and thus a missing field deserializes to an empty array.